engine-comparison/begin-experiment.sh

```diff
@@ -5,59 +5,69 @@
 . $(dirname $0)/../common.sh
 . ${SCRIPT_DIR}/common-harness.sh
 
+[[ -z $2 ]] && echo "Warning: Usage: Please specify benchmarks, as well as at least one fuzzing engine"
+
 DD=$(date +%d)
 MM=$(date +%m)
 INSTANCE_NAME="dispatcher-${DD}-${MM}"
 # DISPATCHER_IMAGE_FAMILY=${DISPATCHER_IMAGE_FAMILY:-"ubuntu-1604-lts"} # Maybe container optimized
 # export PROJECT_NAME="google.com:fuzz-comparisons"
 
-# These will frequently/usually be defined by the user
-export JOBS=${JOBS:-8}
-export N_ITERATIONS=${N_ITERATIONS:-5}
-
-# Define $BENCHMARKS
-if [[ $1 == 'all' ]]; then
-  for b in $(find ${SCRIPT_DIR}/../*/build.sh -type f); do
-    BENCHMARKS="$BENCHMARKS $(basename $(dirname $b))"
-  done
-#elif [[ $1 == 'other alias' ]]; do
-else
-  BENCHMARKS=$(echo $1 | tr ',' ' ')
-fi
-
-echo "Restarting gcloud instance. Instance should already be created (with gcloud_creator.sh)"
 # Start one gcloud instance for the dispatcher
-gcloud compute instances start $INSTANCE_NAME --zone=$GCLOUD_ZONE
-
-# Can't ssh immediately
-while :
-do
-  sleep 6 # arbitrary
-  ! (gcloud compute ssh $INSTANCE_NAME --command="mkdir ~/input" --zone=$GCLOUD_ZONE \
-    2>&1 | grep "ERROR") && break # Break loop as soon as there is no error
-  echo "GCloud VM isn't ready yet. Rerunning SSH momentarily"
-done
+# echo "Restarting gcloud instance. Instance should already be created (with gcloud_creator.sh)"
+create_or_start $INSTANCE_NAME
+robust_begin_gcloud_ssh $INSTANCE_NAME
+gcloud compute ssh $INSTANCE_NAME --command="mkdir ~/input" --zone=$GCLOUD_ZONE
 
 # Send configs for the fuzzing engine
 FENGINE_CONFIGS=${@:2}
 
-if [[ -d tmp-configs ]]; then
-  rm -r tmp-configs
+if [[ -d fengine-configs ]]; then
+  rm -r fengine-configs
 fi
-mkdir tmp-configs
+mkdir fengine-configs
 for FENGINE in $FENGINE_CONFIGS; do
-  cp $FENGINE tmp-configs/$FENGINE
+  cp $FENGINE fengine-configs/$FENGINE
 done
-gcloud compute scp tmp-configs/ ${INSTANCE_NAME}:~/input --recurse --zone=$GCLOUD_ZONE
-rm -r tmp-configs
+
+gcloud compute scp fengine-configs/ ${INSTANCE_NAME}:~/input --recurse --zone=$GCLOUD_ZONE
+rm -r fengine-configs
+
+# These will frequently/usually be defined by the user
+JOBS=${JOBS:-8}
+N_ITERATIONS=${N_ITERATIONS:-5}
+
+# Send configs
+AUTOGEN=${SCRIPT_DIR}/autogen
+
+if [[ ! -d $AUTOGEN ]]; then
```

Comment:
Yeah, going back to my first comment^, maybe just my next PR should restructure all config. This code works for now, though.